### PR TITLE
Add legal name 11 21 2022

### PR DESCRIPTION
**Pull request by InternetRamen**
Solves https://hackclub.slack.com/archives/CN523HLKW/p1627483455259900?thread_ts=1627483333.258800&cid=CN523HLKW

This PR:
 - adds `legal_name` to the `users` table
 - adds a hidden input field to user settings
 - adds `class_controller.js`, a controller that toggles classes

**Best reviewed by any interaction with Stripe API and payments**
Payments may not work with legal names. I have not tested.

**Screenshots**
chrome_GXWqynV5DX
image

### CODE REVIEW

**Comment by maxwofford** [comment]
Love this! It's been a requested feature by so many people

**Comment by InternetRamen** [comment]
Need some help resolving the merge conflicts: 
image

Which version should I merge?

**Comment by InternetRamen** [comment]
Changes to UI 
 - [x] Button that reveals legal name field  
 Mechanics
 - [x] Use legal name only if legal name has a value

**Comment by InternetRamen** [comment]
I'm unable to figure out how to change StripeCard `stripe_name` field. This field is the one that shows up on the cards page. I'm not sure if stripe_name is even the name legally associated with the card.
image

I only changed legal name to "sfv fv" once and then changed it to "" again, yet all of them are "sfv fv"

Will ask Caleb next week :)

**Comment by InternetRamen** [comment]
Should be good to go 👍

**Comment by garyhtou** [comment]
just a thought (i haven't looked through the full pr, so maybe I don't have the full context): since the `full_name` field (usually) already contains legal names in the database, would it make sense to add a `display_name` field instead? That `display_name` field would be optional and defaults to `nil`.

**Comment by InternetRamen** [comment]
That makes a good point but I'm not sure if people set their current name as their legal name or their display name

**Comment by cjdenio** [comment]
@garyhtou since most places in the UI use the display name, we figured it made sense to consider `full_name` the display name.

also if you don't have a legal name set, it considers `full_name` _both_ your legal and display name

File: app/models/user.rb [hunk 1]
```diff
@@ -81,12 +82,32 @@ class User < ApplicationRecord
...
   before_save :on_phone_number_update
 
+  after_update do
+    if stripe_cardholder.present?
+      name = (legal_name.presence || full_name).split(" ")
+      if full_name_previously_changed? || legal_name_previously_changed?
+        StripeService::Issuing::Cardholder.update(stripe_cardholder.stripe_id, {
+                                                    individual: {
+                                                      first_name: name.first,
+                                                      last_name: name[1..name.length].join(" "),
```
Comment (cjdenio) [inline app/models/user.rb]:
```suggestion
                                                      last_name: name[1..].join(" "),
```

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -25,11 +25,23 @@
             <%= form.label :seasonal_themes_enabled, "#{by_season "✨", override_preference: true, fall: "🎃", winter: "🎄"} Enable #{current_season(override_preference: true)} theme", class: "h3 bold" %>
         </div>
         <% end %>
-
-        <div class="field">
-        <%= form.label :full_name %>
-        <%= form.text_field :full_name, required: true, placeholder: "Fiona Hackworth", pattern: '^\S+.+' %>
+        <div data-controller="class" data-class-class-name-value="display-none">
+            <div class="field">
+            <%= form.label :full_name, "Display Name" %>
```
Comment (cjdenio) [inline app/views/users/edit.html.erb]:
```suggestion
            <%= form.label :full_name, "Display name" %>
```

**Review by cjdenio — CHANGES_REQUESTED** [review]

**Comment by InternetRamen** [comment]
- [x] Change Display Name to Full Name when Legal Name dropdown is not shown

**Comment by InternetRamen** [comment]
chrome_GZtXbveXuQ

**Comment by InternetRamen** [comment]
This PR is good to go

**Comment by cjdenio** [comment]
Closing for now because this is inactive and needs some more work.

**Pull request closed without merging**